File: open_days_frontend/lib/models/event_response_model.dart
import 'package:open_days_frontend/constants/constants.dart';

class EventResponseModel {
  bool isOnline;
  String location;
  String dateTime;
  String meetingLink;
  String organizerId;
  String activityName;
  String organizerLastName;
  String organizerFirstName;
  String operationResult = operationResultFailure;

  EventResponseModel({
    this.isOnline = false,
    this.location = '',
    this.dateTime = '',
    this.meetingLink = '',
    this.organizerId = '',
    this.activityName = '',
    this.organizerLastName = '',
    this.organizerFirstName = '',
  });

  factory EventResponseModel.fromJson(Map<String, dynamic> json) {
    return EventResponseModel(
      isOnline: json['isOnline'] ?? false,
      location: json['location'] ?? '',
      dateTime: json['dateTime'] ?? '',
      meetingLink: json['meetingLink'] ?? '',
      organizerId: json['organizerId'] ?? '',
      activityName: json['activityName'] ?? '',
      organizerLastName: json['organizerLastName'] ?? '',
      organizerFirstName: json['organizerFirstName'] ?? '',
    );
  }
}

File: open_days_backend/src/main/java/com/sapientia/open/days/backend/ui/model/response/EventResponseModel.java
package com.sapientia.open.days.backend.ui.model.response;

import java.io.Serial;

@SuppressWarnings("unused")
public class EventResponseModel {
	private long id;
	private boolean isOnline;
	private String location;
	private String dateTime;
	private String meetingLink;
	private String organizerId;
	private String activityName;
	private String organizerLastName;
	private String organizerFirstName;

	@Serial
	private static final long serialVersionUID = 3667708176484458123L;

	public long getId() {
		return id;
	}

	public boolean isIsOnline() {
		return isOnline;
	}

	public String getLocation() {
		return location;
	}

	public String getDateTime() {
		return dateTime;
	}

	public String getMeetingLink() {
		return meetingLink;
	}

	public String getOrganizerId() {
		return organizerId;
	}

	public String getActivityName() {
		return activityName;
	}

	public String getOrganizerLastName() {
		return organizerLastName;
	}

	public String getOrganizerFirstName() {
		return organizerFirstName;
	}

	public void setId(long id) {
		this.id = id;
	}

	public void setIsOnline(boolean isOnline) {
		this.isOnline = isOnline;
	}

	public void setLocation(String location) {
		this.location = location;
	}

	public void setDateTime(String dateTime) {
		this.dateTime = dateTime;
	}

	public void setMeetingLink(String meetingLink) {
		this.meetingLink = meetingLink;
	}

	public void setOrganizerId(String organizerId) {
		this.organizerId = organizerId;
	}

	public void setActivityName(String activityName) {
		this.activityName = activityName;
	}

	public void setOrganizerLastName(String organizerLastName) {
		this.organizerLastName = organizerLastName;
	}

	public void setOrganizerFirstName(String organizerFirstName) {
		this.organizerFirstName = organizerFirstName;
	}
}
